src/mpd.c

```diff
@@ -368,31 +434,50 @@ mpd_time(char *buffer, size_t bufferlen, time_t t)
  *
  * @param range the range argument
  * @param start_pos set by this method to the start position
- * @param end_pos set by this method to the end postion
+ * @param end_pos set by this method to the end position
  * @return 0 on success, -1 on failure
+ *
+ * @see https://github.com/MusicPlayerDaemon/MPD/blob/master/src/protocol/RangeArg.hxx
+ *
+ * For "window START:END" The end index can be omitted, which means the range is open-ended.
  */
 static int
-mpd_pars_range_arg(char *range, int *start_pos, int *end_pos)
+mpd_pars_range_arg(const char *range, int *start_pos, int *end_pos)
 {
   int ret;
 
-  if (strchr(range, ':'))
+  static char separator = ':';
+  char* sep_pos = strchr(range, separator);
+
+  if (sep_pos)
     {
-      ret = sscanf(range, "%d:%d", start_pos, end_pos);
-      if (ret < 0)
-	{
-	  DPRINTF(E_LOG, L_MPD, "Error parsing range argument '%s' (return code = %d)\n", range, ret);
-	  return -1;
-	}
+      *sep_pos++ = '\0';
```

Comment:
I've learned in another PR that this is preferred to be two operations instead of one